lib/rest-api.php

```diff
@@ -183,3 +183,38 @@ function gutenberg_auto_draft_get_sample_permalink( $permalink, $id, $title, $na
 	return $permalink;
 }
 add_filter( 'get_sample_permalink', 'gutenberg_auto_draft_get_sample_permalink', 10, 5 );
+
+/**
+ * Filters WP_User_Query arguments when querying users via the REST API.
+ *
+ * Allow using the has_published_post argument.
+ *
+ * @param array           $prepared_args Array of arguments for WP_User_Query.
+ * @param WP_REST_Request $request       The REST API request.
+ *
+ * @return array Returns modified $prepared_args.
+ */
+function gutenberg_rest_user_query_has_published_posts( $prepared_args, $request ) {
+	if ( ! empty( $request['has_published_posts'] ) && true === $request['has_published_posts'] ) {
+		$prepared_args['has_published_posts'] = get_post_types( array( 'show_in_rest' => true ), 'names' );
+	}
+	return $prepared_args;
+}
+add_filter( 'rest_user_query', 'gutenberg_rest_user_query_has_published_posts', 10, 2 );
+
+
+/**
+ * Filters REST API collection parameters for the users controller.
+ *
+ * @param array $query_params JSON Schema-formatted collection parameters.
+ *
+ * @return array Returns the $query_params with "has_published_posts".
+ */
+function gutenberg_rest_user_collection_params_has_published_posts( $query_params ) {
+	$query_params['has_published_posts'] = array(
+		'description' => __( 'Limit result set to users who have published posts.', 'gutenberg' ),
+		'type'        => 'boolean',
```

Comment:
The `WP_Users_Query`'s `has_published_posts` param can be of type boolean or array. I understand we may want to do the same in the REST API. Looked a bit for examples of similar cases but couldn't find any. Are you aware what are our options here?

Reply:
Given how close we are to release, let's just stick with a boolean for now.

Reply:
Done in https://github.com/WordPress/gutenberg/pull/32662/commits/09dce10af7bc70ea94f722a9ff07b39416bee45d, and also changed the latest-posts block's `has_published_posts` to `[ 'post' ]` in https://github.com/WordPress/gutenberg/pull/32662/commits/c54306d11098b9529e24f74a85565e3b01e2ce85 'cause we don't care for other post-types there :+1: